src/cider/nrepl/middleware/util/java/parser.clj

```diff
@@ -243,8 +243,10 @@
 
 (defn source-path
   "Return the relative `.java` source path for the top-level class."
-  [class]
-  (-> (str/replace (str class) #"\$.*" "")
+  [klass]
+  (-> (str klass)
+      (str/replace #"^class " "")
```

Comment:
Btw, I didn't pay much attention to this before, but I don't quite get it. Why would we strip this here instead of on the client-side? Seems a bit odd to be passing either `class Something` or `Something`.

Reply:
```clojure
(io/resource "class java/util/ArrayList.java")
;; => nil
(io/resource "java/util/ArrayList.java")
;; =>
;; #object[java.net.URL
;;         0x69a72033
;;         "file:/home/oleh/git/java/openjvm-8-src/java/util/ArrayList.java"]
(io/resource "java/util/ArrayList.class")
;; =>
;; #object[java.net.URL
;;         0x249e92b1
;;         "jar:file:/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/rt.jar!/java/util/ArrayList.class"]
```

Reply:
I see.